### DISCUSSION TITLE
AraELECTRA for NER

### DISCUSSION THREAD

**Discussion by Astudnew** [aub-mind/arabert#59] [Q&A] [answered]
Hi
how can I use AraELECTRA for named entity recognition?

**Comment by WissamAntoun** [comment, marked as answer]
Use this notebook and just change the model name https://colab.research.google.com/drive/128wX6r_Nz-2LZ_lROeWjWS_yTi96KR7J?usp=sharing

**Reply by Astudnew** [reply]
Thanks a lot.
which model name can i use?when I use model name ""aubmindlab/araelectra-base"" I get this error:

OSError: Can't load config for 'AraELECTRA-base'. Make sure that:

- 'AraELECTRA-base' is a correct model identifier listed on 'https://huggingface.co/models'

- or 'AraELECTRA-base' is the correct path to a directory containing a config.json file

*****************************************************************************************************************

another Q please, in AraElectra, Farasa is used? because in the link you send above you use "ArabertPreprocessor" ?

**Reply by WissamAntoun** [reply]
for the model you need to use `aubmindlab/araelectra-base-discriminator` , and send `aubmindlab/araelectra-base` to the `ArabertPreprocessor`.

Also Araelectra doesn't need Farasa. ArabertPreprocessor will handle it internally

**Reply by WissamAntoun** [reply]
yes, I will probably change it soon to accept the whole model name from huggingface

**Reply by WissamAntoun** [reply]
> yes, I will probably change it soon to accept the whole model name from huggingface

And by soon, I mean i just added the change, no you can provide the same model name from huggingface to the preprocessor

**Reply by Astudnew** [reply]
Thanks, it works but why get this message in training part:
"Some weights of the model checkpoint at aubmindlab/araelectra-base-discriminator were not used when initializing ElectraForTokenClassification: ['discriminator_predictions.dense.weight', 'discriminator_predictions.dense.bias', 'discriminator_predictions.dense_prediction.weight', 'discriminator_predictions.dense_prediction.bias']
- This IS expected if you are initializing ElectraForTokenClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing ElectraForTokenClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Some weights of ElectraForTokenClassification were not initialized from the model checkpoint at aubmindlab/araelectra-base-discriminator and are newly initialized: ['classifier.weight', 'classifier.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference."

**Reply by WissamAntoun** [reply]
ignore it, these are some warnings that you are not re-using the final dense layers used fine-tuning which is expected
